R/EngineRestGitLab.R

```diff
@@ -103,17 +103,26 @@ EngineRestGitLab <- R6::R6Class("EngineRestGitLab",
     #' @param date_from A starting date to look commits for.
     #' @param date_until An end date to look commits for.
     #' @param settings A list of  `GitStats` settings.
+    #' @param .storage A storage of `GitStats` object.
     #' @return A table of commits.
     pull_commits = function(org = NULL,
                             repos = NULL,
                             date_from,
                             date_until = Sys.date(),
-                            settings) {
+                            settings,
+                            .storage = NULL) {
       if (is.null(repos)) {
-        repos_table <- self$pull_repos_supportive(
-          org = org,
-          settings = list(search_param = "org")
-        )
+        if (is.null(.storage$repositories)) {
+          repos_table <- self$pull_repos_supportive(
+            org = org,
+            settings = list(search_param = "org")
+          )
+        } else {
+          repos_table <- .storage$repositories %>%
+            dplyr::filter(
+              organization == org
```

Comment:
What if we have more than one org? Shoulnd't be %in% here?

Reply:
The engine functions are called from the `GitHost` wrappers which iterate over vector of orgs - that is why this functions deal alway with only one `org`. Here is a glimpse from the line 122 of `GitHost.R`:

Reply:
Ok, I get it, then it's fine. Thank you for clarification :)